pkg/bpfdebug/ifindex.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bpfdebug
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cilium/cilium/pkg/lock"
+
+	"github.com/vishvananda/netlink"
+)
+
+type linkMap map[int]string
+
+var (
+	ifindexMap = linkMap{}
+	mutex      lock.RWMutex
+)
+
+func ifname(ifindex int) string {
+	mutex.RLock()
+	defer mutex.RUnlock()
+
+	if name, ok := ifindexMap[ifindex]; ok {
+		return name
+	}
+
+	return fmt.Sprintf("%d", ifindex)
+}
+
+func init() {
+	go func() {
+		for {
+			newMap := linkMap{}
+
+			links, err := netlink.LinkList()
+			if err != nil {
+				goto sleep
+			}
+
+			for _, link := range links {
+				newMap[link.Attrs().Index] = link.Attrs().Name
+			}
+
+			mutex.Lock()
+			ifindexMap = newMap
+			mutex.Unlock()
+
+		sleep:
+			time.Sleep(time.Duration(15) * time.Second)
```

Comment:
Wouldn't be easier to have a function to add/remove interfaces whenever a new endpoint is added or removed?

Reply:
This allows resolving of all interface names including cilium_host, cilium_net, cilium_vxlan, and native interfaces.